ethereumj-core/src/main/java/org/ethereum/publish/LifeCycleSubscription.java

```diff
@@ -0,0 +1,53 @@
+package org.ethereum.publish;
+
+import org.ethereum.publish.event.Event;
+
+import java.util.function.BiConsumer;
+
+/**
+ * Abstraction that holds together event type, event processor (a.k.a consumer) and optional conditions that resolves
+ * event processing and auto-unsubscribe.
+ *
+ * @author Eugene Shevchenko
+ */
+public class LifeCycleSubscription<E extends Event<D>, D> extends Subscription<E, D> {
+
+    public static class LifeCycle {
```

Comment:
Let's make it simple. Each subscription has a lifecycle which just holds flags that user is gonna change. We may just add a check like `lifeCycle.unsubscribed || ...` to `needUnsubscibeAfter` method.

Reply:
Ok, I'll move this functionality to `Subscription`